### PR TITLE
Mp/search sender address

### PR DESCRIPTION
**Pull request by mvpoyatt**
Updated endpoint to support recent packets, tx hashes, sender addresses and packet IDs

## Summary by CodeRabbit

- **New Features**
  - Introduced enhanced packet search functionalities, allowing searches by various criteria such as Tx Hashes, Sender Addresses, and Packet IDs.

- **Improvements**
  - Improved error handling during packet search and retrieval processes.
  - Added more robust and efficient search parameter handling.

- **Updates**
  - Updated `shortenHex` function to handle optional `hex` parameter more gracefully.
  - Added `senderAddress` field to the `Packet` interface for better packet identification.

### CODE REVIEW

File: app/api/packets/route.ts [hunk 1]
```diff
@@ -1,25 +1,69 @@
...
 
-  if (!txHash) {
-    try {
-      return NextResponse.json(await getRecentPackets());
-    } catch (err) {
-      logger.error('Error getting recent packets: ' + err);
+  // Format as strings for graphql query
+  searchValue ? searchValue = '"' + searchValue + '"' : searchValue = '';
+  from ? from = '"' + from + '"' : from = '';
+  to ? to = '"' + to + '"' : to = '';
```
[the remark this answers is not in the document]
Comment (dshiell) [reply app/api/packets/route.ts:22]: i do like this suggestion

File: app/api/packets/helpers.ts [hunk 2]
```diff
@@ -1,8 +1,11 @@
 import { Packet, PacketStates } from 'utils/types/packet';
-import logger from 'utils/logger';
 import { packet } from '../utils/cosmos/polyibc';
+import logger from 'utils/logger';
 
 function stringToState(state: string) {
+  if (!state) {
+    return PacketStates.SENT;
+  }
```
Comment (Inkvi) [inline app/api/packets/helpers.ts:8]: Why do you need this if `default` in switch statement does the same?

Comment (mvpoyatt) [reply app/api/packets/helpers.ts:8]: Good catch, already fixed this somewhere else

File: app/api/packets/helpers.ts [hunk 3]
```diff
@@ -173,55 +149,65 @@
...
+}
+
+export function generateQueryParams(params: QueryFilters): string {
+  let reqParams = '';
+  params.limit = params.limit || 1000;
+  reqParams += `limit: ${params.limit}, `;
+  if (params.offset) reqParams += `offset: ${params.offset}, `;
+  if (params.orderBy) reqParams += `orderBy: ${params.orderBy}, `;
+  if (params.where) reqParams += `where: {${params.where}}`;
+  return '(' + reqParams + ')';
```
[the remark this answers is not in the document]
Comment (mvpoyatt) [reply app/api/packets/helpers.ts:212]: Less clear

File: app/api/packets/helpers.ts [hunk 4]
```diff
@@ -14,10 +14,7 @@
   }
 }
 
-async function processPacketRequest(packetRequest: {
-  query: string
-  variables: { txHash?: string, limit?: number }
-}): Promise<Packet[]>{
+export async function processRequest(packetRequest: { query: string }): Promise<Packet[]>{
```
[the remark this answers is not in the document]
Comment (mvpoyatt) [reply app/api/packets/helpers.ts:17]: Error handling around this

**Review by Inkvi — APPROVED** [review]

**Pull request merged**